File: Demo4Angular/src/app/authority/models/api-resource-request.model.ts
import { RequestModel, PaginatedRequestModel, } from 'src/app/shared/request.model';
import { ApiResource, ApiScope, ApiSecret } from './api-resource.model';

export class ApiResourceRequestModel extends PaginatedRequestModel {
    enabled?: boolean;
    name?: string;
    displayName?: string;
    description?: string;
    apiResource: ApiResource;

    constructor(pageIndex?: number, pageSize?: number, orderBy?: string, direction?: string) {
        super(pageIndex, pageSize, orderBy, direction);
    }
}

export class ApiScopeRequestModel extends RequestModel {
    apiScope: ApiScope;

    constructor() {
        super();
    }
}

export class ApiSecretRequestModel extends PaginatedRequestModel {
    apiSecret: ApiSecret;

    constructor() {
        super();
    }
}

File: Demo4DotNetCore/Demo4DotNetCore.ResourceServer/Areas/Identity/RequestModel/ApiResourceRequestModel.cs
﻿using Demo4DotNetCore.ResourceServer.Identity.Model;
using Demo4DotNetCore.ResourceServer.Model;

namespace Demo4DotNetCore.ResourceServer.Identity.RequestModel
{
    public class ApiResourceRequestModel : PaginatedRequestModel
    {
        public bool? Enabled { get; set; }
        public string Name { get; set; }
        public string DisplayName { get; set; }
        public string Description { get; set; }

        public ApiResource ApiResource { get; set; }
    }

    //public abstract class SecretRequestModel : BaseRequestModel
    //{
    //    public Secret Secret { get; set; }
    //}

    public class ApiSecretRequestModel : BaseRequestModel
    {
        public ApiSecret ApiSecret { get; set; }
    }

    public abstract class UserClaimRequestModel : BaseRequestModel
    {
        public UserClaim UserClaim { get; set; }
    }

    public class ApiScopeClaimRequestModel : UserClaimRequestModel
    {
        public ApiScopeClaim ApiScopeClaim { get; set; }
    }

    public class ApiScopeRequestModel : BaseRequestModel
    {
        public ApiScope ApiScope { get; set; }
    }

    //public class ApiResourceClaimRequestModel : UserClaimRequestModel
    //{
    //    public ApiResourceClaim ApiResourceClaim { get; set; }
    //}

    //public abstract class PropertyRequestModel
    //{
    //    public Property Property { get; set; }
    //}

    //public class ApiResourcePropertyRequestModel : PropertyRequestModel
    //{
    //    public ApiResourceProperty ApiResourceProperty { get; set; }
    //}
}
